packages/purpose-process/src/services/purposeService.ts

```diff
@@ -152,6 +159,47 @@ export function purposeServiceBuilder(
 
       return retrievePurposeVersionDocument(purposeId, version, documentId);
     },
+    async deletePurposeVersion({
+      purposeId,
+      versionId,
+      organizationId,
+      correlationId,
+    }: {
+      purposeId: PurposeId;
+      versionId: PurposeVersionId;
+      organizationId: TenantId;
+      correlationId: string;
+    }): Promise<void> {
+      logger.info(`Deleting Version ${versionId} in Purpose ${purposeId}`);
+
+      const purpose = await retrievePurpose(purposeId, readModelService);
+
+      if (organizationId !== purpose.data.consumerId) {
+        throw organizationIsNotTheConsumer(organizationId);
+      }
+
+      const purposeVersion = retrievePurposeVersion(versionId, purpose);
+
+      if (purposeVersion.state !== purposeVersionState.waitingForApproval) {
```

Comment:
we should also probably check that the requested version is not the only one existing for the purpose.
Because in that case the entire purpose should be deleted using the endpoint `DELETE@/purpose/:purposeId` with the event `WaitingForApprovalPurposeDeleted`

Reply:
Updated